### PR TITLE
WT-6174 fix dhandle->session_inuse counter leak in __evict_walk

### PR DESCRIPTION
**Pull request by igorsol**
In __evict_walk, the dhandle->session_inuse counter is increased and decreased to walk the tree but in case of errors, the code jumps over the code that decreases the counter and basically leaves the dhandle as in-use so it never gets deleted during sweeps.

This patch adds a fix to decrease the counter even in case of errors.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by tetsuo-cpp** [comment]
autobuild: retest this please

**Comment by tetsuo-cpp** [comment]
Hey @igorso, nice to meet you. Thanks for submitting this patch!

Our team usually merges changes into the `develop` branch so I've just moved this PR to target that branch instead.

I'll kick off a build tomorrow to test these changes and I'll get another member of my team review the patch also. But from my point of view, this change looks good!

**Comment by igorsol** [comment]
Hello @tetsuo-cpp . Thank you. Looking forward for the tests result.

**Comment by tetsuo-cpp** [comment]
@igorsol 
I just remembered one other thing. We'll need you to sign our [Contributor's Agreement](https://www.mongodb.com/legal/contributor-agreement) before going forward with this.

**Comment by lukech** [comment]
ok to test

**Comment by lukech** [comment]
evergreen retry

**Comment by lukech** [comment]
test this please

**Comment by igorsol** [comment]
I signed contributor agreement

**Comment by lukech** [comment]
evergreen retry

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -1538,6 +1538,12 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
     if (dhandle_locked)
         __wt_readunlock(session, &conn->dhandle_lock);
 
+    if (incr) {
+        WT_ASSERT(session, dhandle->session_inuse > 0);
+        (void)__wt_atomic_subi32(&dhandle->session_inuse, 1);
+        incr = false;
```
Comment (tetsuo-cpp) [inline src/evict/evict_lru.c:1544]: This value won't get read again so we should remove this.

Comment (igorsol) [reply src/evict/evict_lru.c:1544]: Hi @tetsuo-cpp 
Ok I will remove this `incr = false` line.
I can also rebase my commit onto the `develop` branch so this PR will contain only single commit. Is this ok?

Comment (tetsuo-cpp) [reply src/evict/evict_lru.c:1544]: Thanks, @igorsol! No need to rebase since I can squash and merge via the Github UI.

**Comment by igorsol** [comment]
@tetsuo-cpp 
redundant line removed

**Comment by tetsuo-cpp** [comment]
I just merged your patch then. Thanks for your patience @igorsol!

**Comment by igorsol** [comment]
Thanks!

**Comment by igorsol** [comment]
@tetsuo-cpp I have one more question: I see this patch has been merged into WT3.2.2 as part of the develop branch. Will it be backported to older WT versions? Or I need to make more pull requests for that?

**Comment by lukech** [comment]
@igorsol WT3.2.2 is the planned next WiredTiger release, which will include this change. We normally don't backport changes to older WT versions. We do backport changes to MongoDB stable branches but that's a different story.

**Pull request merged**